src/megatron/bridge/training/utils/weight_decay_utils.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable
+import torch
+
+def get_no_weight_decay_cond(
+    no_weight_decay_cond_type: str, default_skip_embedding_weight_decay: bool
+    ) -> Callable[[str, torch.Tensor], bool]:
+    """Get the no weight decay condition function."""
+
+    # Default case: no_wd_decay_cond_type is None
+    no_weight_decay_cond_fn = None
+
+    if no_weight_decay_cond_type == 'qwen3_next':
```

Comment:
please do not make conditions like this, will be harder to maintain in the future.
you can just rename this function to `qwen3_next_no_weight_decay_cond`, make it only specific for qwen3-next. in the future if there are other models, they can also make their own function, not add more conditions.